proposal.md

```diff
@@ -1,20 +1,58 @@
+
+That being said the whole reason I picked this project over a data visualization is I thought it would be fun to make a highly fault tolerant web game - so I am really leaning that direction even if it takes longer than rendering a simulation locally to a terminal.
 
 ## Outline Structure
 
-> Describe how you'll organize your code. What is the process and
-> supervision structure? If it uses a framework, how does it fit in. I
-> just need to understand the direction you plan to take. Again, three
-> or four sentences are probably enough.
+I've begun to work a bit on the project and I currently have this module structure:
+
+2DShooterPhoenixServer
+-> this will have one or multiple GameServers running on it (last priority is implementing support for concurrent games.)
+
+GameServer
+-> has/supervises multiple child processes
+-> I believe the game server will initially be :permanent in the case of a single game instance, :temporal if I add support for concurrent games.
+-> handles user input and delegates that input to respective servers
+-> handles system time changes and passes that information down to children to maintain synchronization across children
+-> uses Task.async to simultaneously update it's children on each call to "tick"
+-> BulletServer
+-> PlayerServer
+-> still deciding where collision detection should take place.
+
+and the other servers:
+
+BulletServer
+-> maintains the status of all currently existing bullets
+-> cleans bullets out after their "expiration" time.
+-> updates bullets appropriately based on the time passed from the parent
+
+PlayerServer
+-> maintains players state, we should be able to dynamically add players like in agar.io
+-> updates players based on inputs handed down from GameServer
+
+KeyVal
+-> key val pair server, currently using this in my BulletServer but I will refactor this out quickly.  Unneccesary in this case but it may be valuable for the lobbyserver if I implement multiple game support.
 
+current problems:
+Collision detection seems like it's going to be a really expensive operation.  Not sure how this is going to work out but this is the next item on my 'chopping block' so I will know in the near future how hard it will be to keep this reasonable.
+An obvious workaround to this is to cap the number of players per game and spin up extra games when that number is exceeded - also reload times for players are a possible solution to this.  The big O of this is conceivably O(bullets * players + bullets * walls) so if we cap one of those it seems easy to run this every 50ms or so.  Maybe this needs it's own server to perform this detection asyncrously and then we can pass the Server callbacks to fire when a specific type of detection is found - who knows, I anticipate this being the most technically challenging problem.
 
```

Comment:
Maybe think about collisions from the bullet's point of view. You could also keep a central hash keyed by {x,y} and make collision detection O(1).